lib/ripper-tags.rb

```diff
@@ -136,15 +140,35 @@ def self.option_parser(options)
 
   def self.process_args(argv, run = method(:run))
     option_parser(default_options) do |optparse, options|
-      file_list = optparse.parse(argv)
+      invalid_options = []
+      file_list = []
+
+      while argv.size > 0
+        begin
+          file_list = optparse.parse(argv)
+          break
+        rescue OptionParser::InvalidOption => err
+          err.args.each do |bad_option|
+            argv.delete(bad_option)
+            invalid_options << bad_option
+          end
+        end
+      end
+      
+      if invalid_options.size > 0 && !options.ignore_unsupported
+        raise OptionParser::InvalidOption, invalid_options.first
```

Comment:
Display error for first bad option only, not sure if we want this or all of them.